src/NeoSharp.Core/Persistence/IRepositoryOfT.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Linq.Expressions;
+
+namespace NeoSharp.Core.Persistence
+{
+    public interface IRepository<TEntity> 
+        where TEntity : NeoEntityBase
+    {
+        void Create(TEntity entity);
+
+        void Delete(TEntity entity);
+
+        void Update(TEntity entity);
+
+        TEntity GetById(byte[] id);
+
+        ReadOnlyCollection<TEntity> GetAll();
```

Comment:
IReadOnlyCollection?

Reply:
I like that. It's almost the same of the IEnumerable where we don't have stupid warnings saying possible IEnumerable of IEnumerable
and with better protection.